Fix: Issue where cron would end up in nested transaction blocks and unable to escape.

zorbathut opened the pull request.

zorbathut committed:
Fix: Issue where cron would end up in nested transaction blocks and unable to escape.

justcool393 commented on files/commands/cron.py:
```diff
@@ -87,52 +87,54 @@ def _run_tasks(db_session_factory: scoped_session):
 	running task does not lock the entire table for its entire run, which would
 	for example, prevent any statistics about status from being gathered.
 	'''
-	db: Session = db_session_factory()
```
this does different things than what we want. the issue is that the logging seemed to be moved in 4552b817f. the way to get correct session behavior is to disuse `scoped_session` entirely and just make `db_session` a `sessionmaker` (pretty much same interface, but better for more explicit state).

the `with` block just hides the underlying issue here.

TLSM committed:
Use sessionmaker instead of context block

Essentially there were two bugs:
1) The logging lines before explicit `Session.begin` caused the root
   "A transaction is already begun on this Session.", which is
   resolved by moving the logging within the explicit session block,
   as intended.
2) We were reusing a Session across cron runs. The architecture was
   intended to not do so, but we didn't have an actual sessionmaker
   available to cron. We probably ought to get rid of scoped_session
   elsewhere, but baby steps to resolve the immediate issue.

In testing, this works correctly with the fix to `db.begin`, and
tests with deliberately reintroducing the logging bug, the changes
to session creation prevent the every-15sec stack trace spam.
